File: src/main/java/net/allape/models/FileModel.java
package net.allape.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class FileModel implements Serializable {

    /**
     * 路径
     */
    @NotNull
    private String path;

    /**
     * 文件名称
     */
    @NotNull
    private String name;

    /**
     * 是否为文件夹
     */
    private boolean directory = false;

    /**
     * 大小
     */
    private long size = 0L;

    /**
     * 权限
     */
    private int permissions = 0;

    /**
     * 是否为本地文件
     */
    private boolean local = true;

    @Override
    public String toString() {
        return (this.directory ? "📁" : "📃") + " " + this.name;
    }

}

File: src/main/kotlin/net/allape/model/FileModel.kt
package net.allape.model

enum class FileModelType {
    // 文件或文件夹, 是能读到内容的东西
    FILE,
    // 非文件夹, 比如列表分隔符或者返回上一级目录的占位符
    NON_FILE
}

data class FileModel(
    var path: String,
    var name: String,
    var directory: Boolean = false,
    var size: Long = 0L,
    var permissions: Int = 0,
    var local: Boolean = true,
    var type: FileModelType = FileModelType.FILE,
) {
    override fun toString(): String = (if (directory) "📁" else "📃") + " " + name
}
